guides/plugins/plugins/checkout/document/add-custom-document-type.md

```diff
@@ -92,97 +147,196 @@ Afterwards we're inserting the translations, one for German, one for English. Fo
+
+                    continue;
+                }
+
+                /** @var LocaleEntity $locale */
+                $locale = $order->getLanguage()->getLocale();
+
+                $html = $this->documentTemplateRenderer->render(
+                    $template,
+                    [
+                        'order' => $order,
+                        'config' => $config,
+                        'rootDir' => $this->rootDir,
+                        'context' => $context,
+                    ],
+                    $context,
+                    $order->getSalesChannelId(),
+                    $order->getLanguageId(),
+                    $locale->getCode()
+                );
+
+                $doc = new RenderedDocument(
+                    $html,
+                    $number,
+                    $config->buildName(),
+                    $operation->getFileType(),
+                    $config->jsonSerialize(),
+                );
+
+                $result->addSuccess($orderId, $doc);
+            } catch (\Throwable $exception) {
+                $result->addError($orderId, $exception);
+            }
+        }
+
+        return $result;
+    }
+
+    public function getDecorated(): AbstractDocumentRenderer
     {
-        $templatePath = $templatePath ?? self::DEFAULT_TEMPLATE;
-
-        return $this->documentTemplateRenderer->render($templatePath, [
-            'order' => $order,
-            'config' => DocumentConfigurationFactory::mergeConfiguration($config, new DocumentConfiguration())->jsonSerialize(),
-            'rootDir' => $this->rootDir,
-            'context' => $context,
-        ], $context, $order->getSalesChannelId(), $order->getLanguageId(), $order->getLanguage()->getLocale()->getCode());
+        throw new DecorationPatternException(self::class);
     }
 
-    public function getFileName(DocumentConfiguration $config): string
+    private function getNumber(Context $context, OrderEntity $order, DocumentGenerateOperation $operation): string
     {
-        return $config->getFilenamePrefix() . $config->getDocumentNumber() . $config->getFilenameSuffix();
+        return $this->numberRangeValueGenerator->getValue(
+            'document_' . self::TYPE,
+            $context,
+            $order->getSalesChannelId(),
+            $operation->isPreview()
+        );
     }
 }
 ```
 
 {% endcode %}
 
-First of all we're injecting the `rootDir` of the Shopware installation into our generator, since we'll need that for rendering our template, and the `DocumentTemplateRenderer`, which will do the template rendering.
+First we're injecting the `rootDir` of the Shopware installation into our renderer, since we'll need that for rendering our template, and the `DocumentTemplateRenderer`, which will do the template rendering. We also inject `orderRepository` to get all order entity by list of order's id, `documentConfigLoader` to load the document configuration and `numberRangeValueGenerator` to get the document number
```

Comment:
```suggestion
First, we are injecting the `rootDir` of the Shopware installation into our renderer since we will need that for rendering our template, and the `DocumentTemplateRenderer`, which will do the template rendering. We also inject `orderRepository` to get all order entities by list of order's id, `documentConfigLoader` to load the document configuration and `numberRangeValueGenerator` to get the document number
```